google/cloud/ndb/model.py

```diff
@@ -4890,12 +4890,21 @@ def _get_property_for(self, p, indexed=True, depth=0):
             prop = self._fake_property(p, next, indexed)
         return prop
 
+    def _clone_properties(self):
+        """Relocate ``_properties`` from class to instance.
+
+        Internal helper, in case properties need to be modified for an instance but not
+        the class.
+        """
+        cls = type(self)
+        if self._properties is cls._properties:
+            self._properties = dict(cls._properties)
+
     def _fake_property(self, p, next, indexed=True):
         """Internal helper to create a fake Property. Ported from legacy datastore"""
         # A custom 'meaning' for compressed properties.
         _MEANING_URI_COMPRESSED = "ZLIB"
-        if hasattr(self, "_clone_properties"):
-            self._clone_properties()
+        self._clone_properties()
```

Comment:
what's the rationale for un-guarding access here? I don't quite understand how this PR is 'reimplementing'

Reply:
It's reimplementing in the sense that it's being copied over from legacy NDB ([see here](https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/model.py#L3255)). It would appear it's absence was an oversight from porting earlier.

We can remove the guard because now we know it's going to be present. The guard was initially present because `_fake_property` had been ported earlier but we didn't have an implementation for `_clone_properties` at the time, so Carlos wasn't sure what to do with that.